carina-utils/src/main/java/com/qaprosoft/carina/core/foundation/utils/naming/TestNamingUtil.java

```diff
@@ -200,12 +201,29 @@ public static String appendTestMethodName(String testName, ITestNGMethod m) {
 
         if (invocationID != -1) {
             // TODO: analyze if "InvCount=nnnn" is already present in name and don't append it one more time
-            testName = testName + " - " + m.getMethodName() + String.format(SpecialKeywords.INVOCATION_COUNTER, String.format("%04d", invocationID));
+            testName = testName + " - " + figureOutTestName(m) + String.format(SpecialKeywords.INVOCATION_COUNTER, String.format("%04d", invocationID));
         } else {
-            testName = testName + " - " + m.getMethodName();
+            testName = testName + " - " + figureOutTestName(m);
         }
 
         return StringEscapeUtils.escapeHtml4(testName);
     }
 
+    private static String figureOutTestName(ITestNGMethod m) {
```

Comment:
that's mine vision of flexible method naming... also constants "{method_name}" etc can be declared in SpecialKeywords.

	private static String adjustTestName(ITestNGMethod m) {
		String testName = Configuration.get(Parameter.TEST_NAMING_PATTERN);
		testName.replace("{method_name}", m.getMethodName());
		testName.replace("{method_description}", m.getDescription());
		testName.replace("{method_priority}", String.valueOf(m.getPriority()));
		testName.replace("{method_thread_pool_size}", String.valueOf(m.getThreadPoolSize()));
		return testName;
	}

As result engineers can define any formatting for example:
test_naming_patter={method_name} - {method-description}
test_naming_patter={method_priority} | {method_name} - {method-description} | {method_thread_pool_size}
etc